### PR TITLE
FEAT: Support read parquet from http urls

### LINKED ISSUE #571: ENH:  support http protocol for `pd.read_parquet`
### Is your feature request related to a problem? Please describe

Now, xorbits does not support http protocol for `read_parquet`, we can support it. A list can be passed to read multiple parquet files.

### PR DESCRIPTION
**Pull request by aresnow1**
## What do these changes do?

Support read http urls for `pd.read_parquet`.

## Related issue number

Fixes #571 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

**Review by qinxuye — APPROVED** [review]
LGTM

File: python/xorbits/_mars/dataframe/datasource/read_parquet.py [hunk 1]
```diff
@@ -495,6 +520,14 @@ def _pandas_read_parquet(cls, ctx: dict, op: "DataFrameReadParquet"):
         out = op.outputs[0]
         path = op.path
 
+        if op.is_http_url:
+            ctx[out.key] = pd.read_parquet(
+                op.path,
+                columns=op.columns,
+                engine=op.engine,
+                **op.read_kwargs or dict(),
```
Comment (codingl2k1) [inline python/xorbits/_mars/dataframe/datasource/read_parquet.py:528]: The `use_arrow_dtype` should be handled here. If `use_arrow_dtype` is True, `pd.read_parquet` use a arrow type mapping.

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource_execution.py [hunk 2]
```diff
@@ -1388,6 +1389,73 @@ def test_read_parquet_fast_parquet(setup):
...
+def test_read_parquet_with_http_url(setup, start_http_server):
+    df, urls = start_http_server
+    mdf = md.read_parquet(urls).execute().fetch()
+    pd.testing.assert_frame_equal(df, mdf)
+
+    mdf1 = md.read_parquet(urls[:1]).execute().fetch()
+    pd.testing.assert_frame_equal(df.iloc[:50], mdf1)
+
+    mdf2 = md.read_parquet(urls[1:]).execute().fetch()
+    pd.testing.assert_frame_equal(df[50:], mdf2)
```
Comment (codingl2k1) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource_execution.py:1456]: Can you add a UT for read_parquet with http url and `use_arrow_dtype` is True?

**Review by codingl2k1 — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**